pymc_marketing/product_incrementality/mv_its.py

```diff
@@ -0,0 +1,445 @@
+                This means that the reduction in sales of existing products will equal
+                the increase in sales of the new product, such that the total sales
+                remain constant."""
+                alpha = np.full(len(coords["background_product"]), alpha_background)
+                beta = pm.Dirichlet("beta", a=alpha, dims="background_product")
+            else:
+                """We assume the market is not saturated. The sum of the beta's will be
+                less than 1. This means that the reduction in sales of existing products
+                will be less than the increase in sales of the new product."""
+                alpha_all = np.full(len(coords["all_sources"]), alpha_background)
+                beta_all = pm.Dirichlet("beta_all", a=alpha_all, dims="all_sources")
+                beta = pm.Deterministic(
+                    "beta", beta_all[:-1], dims="background_product"
+                )
+                pm.Deterministic("new sales", beta_all[-1])
+
+            # expectation
+            mu = pm.Deterministic(
+                "mu",
+                intercept[None, :] - treatment_sales[:, None] * beta[None, :],
+                dims=("time", "background_product"),
+            )
+
+            # likelihood
+            normal_dist = pm.Normal.dist(mu=mu, sigma=sigma)
+            pm.Truncated(
+                "y",
+                normal_dist,
+                lower=0,
+                observed=_existing_sales,
+                dims=("time", "background_product"),
+            )
+
+        return model
+
+    def sample_prior_predictive(self):
+        """Sample from the prior predictive distribution."""
+        with self.model:
+            self.idata = pm.sample_prior_predictive(random_seed=self.rng)
+
+    def fit(self):
+        """Fit the model to the data."""
+        with self.model:
+            self.idata.extend(pm.sample(**self.sample_kwargs, random_seed=self.rng))
+
+    def sample_posterior_predictive(self):
+        """Sample from the posterior predictive distribution."""
+        with self.model:
+            self.idata.extend(
+                pm.sample_posterior_predictive(
+                    self.idata,
+                    var_names=["mu", "y"],
+                    random_seed=self.rng,
+                )
+            )
+
+    def calculate_counterfactual(self):
+        """Calculate the counterfactual scenario of never releasing the new product."""
+        zero_sales = np.zeros(self.data[self.treatment_sales].shape, dtype=np.int32)
+        self.counterfactual_model = pm.do(self.model, {"treatment_sales": zero_sales})
+        with self.counterfactual_model:
+            self.idata.extend(
+                pm.sample_posterior_predictive(
+                    self.idata,
+                    var_names=["mu", "y"],
+                    random_seed=self.rng,
+                    predictions=True,
```

Comment:
If you want to expose the counterfactual to the users and find `model.idata.predictions` confusing, then we can explore a property to expose it via a different name. What's your pulse on the usage and users?